examples/stringification/main.cpp

```diff
@@ -0,0 +1,491 @@
+
+    {
+        std::valarray<int> value { 7, 3, 2 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::initializer_list<int> value { 1, 7, 0 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::vector<int> value { 4, 2, 7 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::deque<int> value { 1, 1, 6, 2 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::list<int> value { 7, 5, 2, 1 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::forward_list<int> value { 7, 0, 3, 7, 4 };
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::stack<int> value {{ 1, 2, 3, 4, 5 }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        std::queue<int> value {{ 1, 2, 3, 4, 5 }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+
+    {
+        const auto data = { 1, 8, 5, 6, 3 };
+
+        std::priority_queue<int> value { data.begin(), data.end() };
+        CHECK_EQ(toString(value), "{?}");
+    }
+    
+    {
+        std::set<int> value {{ 1, 2, 2, 6, 1, 4 }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+    
+    {
+        std::multiset<int> value {{ 1, 2, 2, 6, 1, 4 }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+    
+    {
+        std::map<int, std::string> value {{ { 1, "foo" }, { 2, "bar" }, { 1, "baz" } }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+    
+    {
+        std::multimap<int, std::string> value {{ { 1, "foo" }, { 2, "bar" }, { 1, "baz" } }};
+        CHECK_EQ(toString(value), "{?}");
+    }
+#else
+    add_stub_asserts(17);
+#endif  // CXX_STANDARD >= 11
+
+#if CXX_STANDARD >= 14
+    {
+        std::integer_sequence<int, 9, 2, 5, 1, 9, 1, 6> value {};
+        CHECK_EQ(toString(value), "{?}");
+    }
+#else
+    add_stub_asserts(1);
+#endif  // CXX_STANDARD >= 14
+
+#if CXX_STANDARD >= 17
+    {
+        const std::string s = "Hello world!";
+        std::string_view value{ s };
+        value = value.substr(0, 5);
+        CHECK_EQ(toString(value), "Hello"); // standard ostream operator is used
+    }
+    
+    {
+        CHECK_EQ(toString(std::nullopt), "{?}");
+        std::optional<int> value {};
+        CHECK_EQ(toString(value), "{?}");
+        value = 9;
+        CHECK_EQ(toString(value), "{?}");
+    }
+    
+#if !DOCTEST_MSVC || (DOCTEST_MSVC >= DOCTEST_COMPILER(19, 29, 0)) // MSVC v141 has defect is default constructable std::variant
+    {
+        std::variant<std::monostate, int, bool> value {};
+        CHECK_EQ(toString(value), "{?}"); // std::monostate
+        value = 5;
+        CHECK_EQ(toString(value), "{?}");
+        value = true;
+        CHECK_EQ(toString(value), "{?}");
+    }
+#else
+    add_stub_asserts(3);
+#endif
+
+#else
+    add_stub_asserts(7);
+#endif  // CXX_STANDARD >= 17
+
+#if CXX_STANDARD >= 20
+    {
+        std::array<int, 3> ar {{ 9, 1, 5 }};
+        std::span<int> value { ar };
+        CHECK_EQ(toString(value), "{?}");
```

Comment:
I want to merge the `string-fix` branch into dev for 2.4.9, while I would like to postpone the stl stringification extension header (for which tests already exist) to 2.5, so either this PR will need to be postponed until 2.5 or you should split it into two parts

Reply:
I still see no problems here. We check that *currently* stl stringification is absent. All is fine. When it is added, this test will be changed with the actual stringification result.

Reply:
Testing that something is absent seems like nothing but bloat to me. I'd really like to have everything but the working tests (so that includes everything relating to a specific standard version) completely removed from this PR before merging. If you don't want to put in the work for it then I'd happily do it for you.